DownFall/Tutorial/TutorialTileCreator.swift

```diff
@@ -24,30 +24,47 @@
 import GameplayKit
 
 struct TutorialTileCreator: TileStrategy {
+    func goldDropped(from monster: EntityModel) -> Int {
+        return 5
+    }
+    
     var updatedEntity: EntityModel?
     var difficulty: Difficulty
-    var entities: [EntityModel]
+    var entities: EntitiesModel
+    var level: Level?
     var randomSource: GKLinearCongruentialRandomSource = GKLinearCongruentialRandomSource()
     
-    init(_ entities: [EntityModel], difficulty: Difficulty, updatedEntity: EntityModel?) {
+    init(_ entities: EntitiesModel, difficulty: Difficulty, updatedEntity: EntityModel?, level: Level?) {
         self.entities = entities
         self.difficulty = difficulty
         self.updatedEntity = updatedEntity
+        self.level = level
     }
     
-    func board(_ boardSize: Int, difficulty: Difficulty) -> [[Tile]] {
+    func board(difficulty: Difficulty) -> [[Tile]] {
+        guard let playerData = playerEntityData else { fatalError("We must have player data to continue") }
+        
         let greenRow = Array(repeating: Tile.greenRock, count: 4)
+        let purpleRow = Array(repeating: Tile.purpleRock, count: 4)
         
         var tiles: [[Tile]] = []
-        switch difficulty {
-        case .easy, .normal, .hard:
+        switch level?.type {
+        case .first, .second, .third, .boss, .none:
             fatalError("You cant create a tutorial board with one of these diffculties")
         case .tutorial1:
             tiles = [
-                [.greenRock, .gold, .greenRock, .greenRock],
+                [.greenRock, .gem, .greenRock, .greenRock],
```

Comment:
This probably should not be  a green rock anymore since I am turning those into "special rocks"